lambdas/check-existing-identity/src/main/java/uk/gov/di/ipv/core/checkexistingidentity/CheckExistingIdentityHandler.java

```diff
@@ -599,4 +605,23 @@ private void sendProfileMatchedAuditEvent(
                         new AuditRestrictedDeviceInformation(deviceInformation));
         auditService.sendAuditEvent(auditEvent);
     }
+
+    private boolean isReprovingWithF2f(
+            CriResponseItem f2fRequest, VerifiableCredentialBundle vcBundle) {
+        // does the user have a F2F response item that was created in response to an intervention,
+        // and they're returning to core with a pending identity
+        return f2fRequest != null && f2fRequest.isReproveIdentity() && vcBundle.isPendingIdentity();
+    }
+
+    private void removeReproveIdentityFlag(CriResponseItem f2fResponseItem) {
```

Comment:
Do we still need to do this if we're enforcing that it's only skipped on the first return after a F2F reprove?

Reply:
gah. I'm not sure why my brain is struggling with this so much...

I suppose we only care if the user manages to achieve a profile - as we wouln't want any further interventions to be skipped. And if they *do* achieve a profile, then they will no longer have a pending identity. So won't skip...

And if they *don't* achieve a profile, then they'll be on an identity proving journey anyway.

So no?